CHANGELOG.md

```diff
@@ -7,10 +7,402 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+                -   accepts `clientType` to determine the matching config
+                -   One of redirectURIInfo (for code flow) or oAuthTokens (for token flow) is required
+        -   Updated `appleRedirectHandlerPOST`
+            -   to accept all the form fields instead of just the code
+            -   to use redirect URI encoded in the `state` parameter instead of using the websiteDomain config.
+            -   to use HTTP 303 instead of javascript based redirection.
+    -   Session recipe changes
+        -   Added mandatory `tenantId` as first parameter to the following recipe exposed functions:
+            -   `createNewSession`
+            -   `createNewSessionWithoutRequestResponse`
+            -   `validateClaimsInJWTPayload`
+        -   Added mandatory `tenantId` in the input for the following recipe interface functions. If any of these functions are overridden, they need to be updated accordingly:
+            -   `createNewSession`
+            -   `getGlobalClaimValidators`
+        -   Added `tenantId` and `revokeAcrossAllTenants` params to `revokeAllSessionsForUser` in the recipe interface.
+        -   Added `tenantId` and `fetchAcrossAllTenants` params to `getAllSessionHandlesForUser` in the recipe interface.
+        -   Added `getTenantId` function to `SessionContainerInterface`
+        -   Added `tenantId` to `fetchValue` function in `PrimitiveClaim`, `PrimitiveArrayClaim`.
+    -   UserRoles recipe changes
+        -   Added mandatory `tenantId` as first parameter to the following recipe exposed functions:
+            -   `addRoleToUser`
+            -   `removeUserRole`
+            -   `getRolesForUser`
+            -   `getUsersThatHaveRole`
+        -   Added mandatory `tenantId` in the input for the following recipe interface functions. If any of these functions are overridden, they need to be updated accordingly:
+            -   `addRoleToUser`
+            -   `removeUserRole`
+            -   `getRolesForUser`
+            -   `getRolesForUser`
+    -   Similar changes in combination recipes (thirdpartyemailpassword and thirdpartypasswordless) have been made
+
+### Changes
+
+-   Recipe function changes:
+    -   Added optional `tenantIdForPasswordPolicy` param to `EmailPassword.updateEmailOrPassword`, `ThirdPartyEmailPassword.updateEmailOrPassword`
+    -   Added optional param `tenantId` to `Session.revokeAllSessionsForUser`. If tenantId is undefined, sessions are revoked across all tenants
+    -   Added optional param `tenantId` to `Session.getAllSessionHandlesForUser`. If tenantId is undefined, sessions handles across all tenants are returned
+-   Adds optional param `tenantId` to `getUserCount` which returns total count across all tenants if not passed.
+-   Adds protected prop `tId` to the accessToken payload
+-   Adds `includesAny` claim validator to `PrimitiveArrayClaim`
+
 ### Fixes
 
 -   Fixed an issue where certain Dashboard API routes would return a 404 for Hapi
 
+### Migration
```

Comment:
do not make it 1, 2, 3.. make it a `####`

Reply:
Done